ztp/siteconfig-generator/siteConfig/filesHandler.go

```diff
@@ -0,0 +1,67 @@
+package siteConfig
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+func GetFiles(path string) ([]os.FileInfo, error) {
+	fileInfo, err := os.Stat(path)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if fileInfo.IsDir() {
+		return ioutil.ReadDir(path)
+	}
+
+	return []os.FileInfo{fileInfo}, nil
+}
+
+func ReadFile(filePath string) ([]byte, error) {
+	return ioutil.ReadFile(filePath)
+}
+
+func ReadExtraManifestResourceFile(fileName string) ([]byte, error) {
+	var dir = ""
+	var err error = nil
+	var ret []byte
+
+	ex, err := os.Executable()
+	if err != nil {
+		return nil, err
+	}
+	dir = filepath.Dir(ex)
+	ret, err = ReadFile(dir + "/" + localExtraManifestPath + "/" + fileName)
+
+	// added fail safe for test runs as `os.Executable()` will fail for tests
+	if err != nil {
+
+		dir, err = os.Getwd()
+		if err != nil {
+			return nil, err
+		}
+		ret, err = ReadFile(dir + "/" + localExtraManifestPath + "/" + fileName)
+	}
+	return ret, err
+}
+
+func GetExtraManifestResourceFiles() ([]os.FileInfo, string, error) {
+	ex, err := os.Executable()
+	if err != nil {
+		return nil, "", err
+	}
+
+	dir := filepath.Dir(ex)
+	files, err := GetFiles(dir + "/" + localExtraManifestPath)
```

Comment:
The code to create the localExtraManifestPath duplicates the same logic above. Can you factor this out into a common function? That makes it easier in the future to add more complex path searching or override via env var, etc

Reply:
Yes there are 2 different returns here but this same logic is above in the ReadExtraManifestResourceFile too. Seems like both methods are solving the same problem of finding a resource file in the same location.